R/assign.R

```diff
@@ -0,0 +1,473 @@
+#' tracks individual plants through time
+#'
+#' @return
+#' @export
+#'
+#' @examples
+
+# required packages -------------------------------------------------------
+library(sf)
+library(mapview) #don't actually need for function, just for checking
+library(dplyr)
+library(tidyr)
+
+# example input data ------------------------------------------------------
+# grasslandData (or exact same format), subset to a unique site, quad,
+# species
+load("./data/grasslandData.rda")
+load("./data/grasslandInventory.rda")
+# prepares the dataset to feed into the 'assign' function (the 'Assign'
+# function will do this ahead of time when the user calls it)
+sampleDat <- grasslandData[grasslandData$Site == "CO"
+                           & grasslandData$Quad == "unun_11"
+                           & grasslandData$Species == "Bouteloua gracilis",]
+# this should be a data.frame
+
+# get the appropriate grasslandInventory data for the "unun_11" quadrat,
+# to tell the 'assign' function when the quadrat was sampled
+sampleInv <- grasslandInventory[["unun_11"]]
+# this should be an integer vector
+
+# 'assign' function ---------------------------------------------------------
+
+assign <- function(sampleDat, inv, dorm, buff, buffGenet, clonal,...){
+  # arguments ---------------------------------------------------------------
+  ## double check the format of the inputs, and add additional columns required
+  dat <- st_sf(sampleDat) # data in 'grasslandData' format, must be an sf object
+    ## add columns for trackID, age, size_t+1, and recruit
+    dat$trackID <- NA
+    dat$age <- NA
+    dat$size_tplus1 <- NA
+    dat$recruit <- NA
+    dat$survives_tplus1 <- NA
+    dat$index <- c(1:nrow(dat))## assign an arbitrary, unique index number to
+    # each row in the dataset
+    dat$ghost = NA
+    dat$trackID = NA
+    dat$rametArea = NA
+  inv <- sort(sampleInv) ## integer vector of quadrat sampling years in
+  # sequential order
+  ## make sure that the 'assignOut' data.frame that contains the output is empty
+  rm(assignOut)
+  ## user-defined arguments
+  dorm <- 0 ## dormancy allowed by the function
+  buff <- .05 ## buffer of movement allowed from year to year, in meters
+  buffGenet <- 0.001 ## buffer between polygons (/2) that is the maximum allowed for
+  # them to be considered genets
+  # overlap <- .50 ## the percentage of overlap (in decimal form) between focal
+  # indiv. and next year indiv. that will be required to consider them both
+  # the same individual (not 100% sure on this one yet...)
+  clonal <- 0 ## binary option that indicates whether this species is allowed to
+  # break into multiple polygons for one individual
```

Comment:
From lines 51 to 60 this looks like code for debugging. Should it be commented out for real use of the function?

Reply:
Yes!